src/prefect/engine.py

```diff
@@ -772,11 +807,13 @@ async def resume_flow_run(flow_run_id):
     if not flow_run.state.is_paused():
         raise NotPausedError("Cannot resume a run that isn't paused!")
 
-    await client.set_flow_run_state(
+    response = await client.resume_flow_run(
         flow_run_id,
-        Running(name="Resuming"),
     )
 
+    if response.status == SetStateStatus.REJECT:
+        raise FlowPauseTimeout("Flow run can no longer be resumed.")
```

Comment:
Per the above on ABORT, this could be a REJECT with a FAILED state to distinguish.